### PR TITLE
Fixed support for row and columns headers in `parseTable` utility

### LINKED ISSUE #8041: [8.0.0+] `toHTML()` returns an error if only `colHeaders` or `rowHeaders` are on
### Description

The `toHTML()` returns an error if `colHeaders` are on in versions 8+

### Steps to reproduce

1. Go to https://jsfiddle.net/handsoncode/bqepxasw/ (7.4.2.)
2. Log result 
image
3. Go to https://jsfiddle.net/handsoncode/3mg8rsxc/ (8.0.0)
4. Log result
image

### Details
* it works well if we turn off both - column and row headers https://jsfiddle.net/handsoncode/021Lp9jd/1/
* it breaks for only `colHeaders` or only `rowHeaders` https://jsfiddle.net/handsoncode/k46vr73o/

### Demo

https://jsfiddle.net/handsoncode/1efLzgu7/ 7.4.2 - last working
https://jsfiddle.net/handsoncode/3mg8rsxc/ 8.0.0 - first broken 
https://jsfiddle.net/handsoncode/6ua5efqr/ - (current) 8.4.0 - still broken

### Your environment
* Handsontable version: 8+
* Browser Name and version: Chrome 89
* Operating System: Windows 10

### PR DESCRIPTION
**Pull request by swistach**
### Context
Parser to generate HTMLTable from Handsontable's instance uses incorrect modifiers to calculate headers offset. Tests were based on headers enabled for rows and column at once what hid the problem.

### How has this been tested?
Initialize Handsontable with *only* one of the headers options:
- `colHeaders`
- `rowHeaders`

Run `toHTML()` method on your Handsontable's instance.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. Fixes #8041

### Affected project(s):
- [x] `handsontable`

### CODE REVIEW

**Review by aninde — APPROVED** [review]
LGTM 👏

File: .changelogs/8041.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "title": "\"Fixed support for row and columns headers in parseTable utility.\"",
```
Comment (jansiegel) [inline .changelogs/8041.json:2]: The additional quotation marks are not needed here, I guess

**Review by jansiegel — DISMISSED** [review]

File: .changelogs/8041.json (hunk 1 above)
Comment (swistach) [reply .changelogs/8041.json:2]: That's something that we suggest in our changelog generator - but right, additional quotation marks are unnecessary 😄

image

Comment (jansiegel) [reply .changelogs/8041.json]: True, true

**Review by jansiegel — APPROVED** [review]

**Review by aninde — APPROVED** [review]
👍

**Pull request merged**